Ah jvb connecting to firebase

AshlingH opened the pull request:
- Added the react-router-dom and firebase NPM packages

- Editted Lib/Firebase.js to export fb as default

- Edited app.js to include a route to /firebase.testing

- Created a firebase.testing.js under pages folder

Page when clicked renders three most recent documents with time.

jendevelops committed:
Add react-router-dom and firebase npm packages

jendevelops committed:
Edit firebase export to have default name fb

jendevelops committed:
Delete placeholder code and add route to FirebaseTesting

jendevelops committed:
Initialize FirebaseTesting page, add onClick func

Created the FirebaseTesting page that renders a hyperlink tag. Tag
assigned an onClick function to write current time of click to Cloud
FireStore.

jendevelops committed:
Expand onClick function to include reading data from cloud FireStore

jendevelops committed:
Add renderClickLog func to conditionally render clickLog local state

jendevelops commented:
Overall, read and write to FireStore was achieved.

Tangential Problem: Having difficulty re-rendering on local state change. Might be something to take a look at together.

laurenmbeatty commented:
@jbatara maybe look at a `useEffect` hook https://reactjs.org/docs/hooks-reference.html. You can have it listening to changes on `clickLog`

jendevelops commented:
> @jbatara maybe look at a `useEffect` hook https://reactjs.org/docs/hooks-reference.html. You can have it listening to changes on `clicksLog`

That's perfect! We'll take a look at that! @AshlingH

dkimlim reviewed (approved):
Code looks good and clear!

jendevelops committed:
Import firebase to FirebaseTest, change currentTime to firebase timestamp

jendevelops committed:
Import useEffect from react, move db read to useEffect

Logic to read from Firestore moved to useEffect. Data read from
clicks collection also reordered by dateTime, and desc; limit is now 5.

jendevelops committed:
Move render function to turnary operator in return

jendevelops commented:
Hi, all! I just added changes to the code to use the useEffect react function. Please let me know if the most recent updates to the code look good!

joeylaguna commented on src/pages/FirebaseTest.js:
```diff
@@ -0,0 +1,46 @@
+import fb from '../lib/firebase';
+import firebase from 'firebase/app';
+
+export default function FirebaseTesting (props) {
```
It looks like `props` is never been used. Do we need to keep it here?

jendevelops replied:
Removed in commit 1a220e5

joeylaguna commented on src/pages/FirebaseTest.js:
```diff
@@ -0,0 +1,46 @@
+import fb from '../lib/firebase';
+import firebase from 'firebase/app';
+
+export default function FirebaseTesting (props) {
```
**Suggestion**: Since this component is for testing, it can totally be changed later, but I think it's worth looking up the ES6 syntax for arrow functions! 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

jendevelops replied:
Changed to ES6 syntax in commit 1a220e5

joeylaguna reviewed (commented):
Great job, everybody!  

Just left a few comments. The one around arrow functions doesn't need to be addressed in this pull request, but I think it will still be very useful to read up on and get used to that syntax.

Once again, great job! Can't wait to see this in action 😄

jendevelops committed:
Remove props, change to es6 arrow notation where needed

jendevelops committed:
Merge branch 'master' into AH-JVB-connecting-to-firebase

jendevelops committed:
Add default argument to useEffect to prevent Firestore read overload

jonbascos commented:
Ugh sorry...got busy. Looks good you two!  Way to work it!
